### PR TITLE
[common] Add support checksum/config annotation to automate roll deployments.

### PR DESCRIPTION
**Pull request by Amos-85**
**Description of the change**
Add support checksum/config annotation to automate roll deployments.

**Benefits**
As long as we change configuration, we want it will take effect right after running helm upgrade and redeploy the pods.

**Possible drawbacks**
N/A

**Applicable issues**

N/A

**Additional information**
[automatically-roll-deployments](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments)

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).

### CODE REVIEW

File: charts/stable/common/README_CHANGELOG.md.gotmpl [hunk 1]
```diff
@@ -9,6 +9,15 @@ All notable changes to this library Helm chart will be documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+### [4.5.0]
+
+#### Added
+
+- Support checksum/config annotations for configMaps to automate roll deployments after config changes.
+
+#### Changed
+
+- Changes in daemonset / deployment & statefulls templates.
```
Comment (bjw-s) [inline charts/stable/common/README_CHANGELOG.md.gotmpl:20]:
```suggestion
```

I don't think there are any other changes to the templates?

File: charts/stable/common/README_CHANGELOG.md.gotmpl [hunk 2]
```diff
@@ -9,6 +9,15 @@ All notable changes to this library Helm chart will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+### [4.5.0]
+
+#### Added
+
+- Support checksum/config annotations for configMaps to automate roll deployments after config changes.
```
Comment (bjw-s) [inline charts/stable/common/README_CHANGELOG.md.gotmpl:16]:
```suggestion
- Support checksum/config annotations for configMaps to automate roll deployments/daemonsets/statefulsets after config changes.
```

**Review by bjw-s — CHANGES_REQUESTED** [review]
Hi @Amos-85, thanks for submitting this PR! I've requested a few small changes.

Could you also maybe add unit tests for these changes (they are under `./helper-charts/common-test/tests/controller`)? Feel free to reach out if you need any help.

**Comment by Amos-85** [comment]
> Hi @Amos-85, thanks for submitting this PR! I've requested a few small changes.
> 
> Could you also maybe add unit tests for these changes (they are under `./helper-charts/common-test/tests/controller`)? Feel free to reach out if you need any help.

@bjw-s , I've add unit tests for the 3 controllers.
It's the first time I used this helm unittest plugin.
Let me know if need to fix something.

**Comment by onedr0p** [comment]
Any reason why you closed this @Amos-85 ?

**Comment by Amos-85** [comment]
@onedr0p,
Sorry, I've committed by mistake wrong changes that I'm taking forward on my fork and I struggle to revert them.
Anyway, I'll create a new PR with a link to the closed one.

**Comment by onedr0p** [comment]
Gotcha, Thanks!

**Pull request closed without merging**